sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -742,12 +750,40 @@ object ScalaJSPluginInternal {
 
   // These settings will be filtered by the stage dummy tasks
   val scalaJSRunSettings = Seq(
+      scalaJSMainEntryPoint := {
+        if (scalaJSUseMainEntryPoint.value) {
+          mainClass.value.fold {
+            throw new MessageOnlyException(
+                "No main class class detected, but scalaJSUseMainEntryPoint " +
+                "was set to true. If you do not want to derive a main entry " +
+                "point from discovered main classes, add " +
+                "`scalaJSMainEntryPoint := None`")
+          } { mainClass =>
+            Some(EntryPoint.mainMethod(mainClass, "main"))
+          }
+        } else {
+          None
+        }
+      },
```

Comment:
Let's just remove it altogether, and inline its value in `scalaJSEntryPoints ++=` just below.